cas-server-core/src/main/java/org/jasig/cas/CentralAuthenticationServiceImpl.java

```diff
@@ -261,8 +262,8 @@ public ServiceTicket grantServiceTicket(
             }
             ticketGrantingTicket.getSupplementalAuthentications().add(currentAuthentication);
         }
-        
-        if (!registeredService.isSsoEnabled() && currentAuthentication == null) {
+
+        if (currentAuthentication == null && !registeredService.getAuthorizationStrategy().isServiceAuthorizedForSso()) {

```

Comment:
Is my diff out of date or did you adjust the name to access strategy but not the accessing methods?